File: typescript/my-first-package/bin/services/myFirstService/dynamic.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
// Module: services.myFirstService.dynamic
const myFirstPackageSchemasMyFirstModuleDynamic = __importStar(require("../../schemas/myFirstModule/dynamic"));
const myFirstPackageServicesMyFirstServiceStructures = __importStar(require("./structures"));
const typelayerCommonDynamic = __importStar(require("typelayer-common"));
function dynamicGetMyFirstFunctionInput(dynamicObject) {
    let constructor;
    let myFirstFunctionInputBuilder;
    let myFirstFunctionInputFieldName;
    let myFirstFunctionInputFieldsLeft;
    let myFirstFunctionInputInit;
    let myFirstFunctionInputMyFirstArgumentValue;
    let myFirstFunctionInputNewFieldsLeft;
    let numFields;
    [constructor, numFields] = dynamicObject.decoderPopStruct("my_first_function_input");
    switch (constructor) {
        case "my_first_function_input":
            myFirstFunctionInputInit = myFirstPackageServicesMyFirstServiceStructures.defaultMyFirstFunctionInput();
            [myFirstFunctionInputFieldsLeft, myFirstFunctionInputBuilder] = [
                numFields,
                myFirstFunctionInputInit
            ];
            while (-1 < myFirstFunctionInputFieldsLeft) {
                if (0 < myFirstFunctionInputFieldsLeft) {
                    myFirstFunctionInputNewFieldsLeft = myFirstFunctionInputFieldsLeft;
                }
                else {
                    myFirstFunctionInputNewFieldsLeft = dynamicObject.decoderPopHasMoreStruct();
                }
                if (0 < myFirstFunctionInputNewFieldsLeft) {
                    myFirstFunctionInputFieldName = dynamicObject.decoderPopField();
                    switch (myFirstFunctionInputFieldName) {
                        case "*":
                            throw new typelayerCommonDynamic.DynamicGetException("Constructor must be first field", { path: dynamicObject.decoderPath() });
                            break;
                        case "my_first_argument":
                            myFirstFunctionInputMyFirstArgumentValue = dynamicObject.decoderPopString();
                            myFirstFunctionInputBuilder.myFirstArgument = myFirstFunctionInputMyFirstArgumentValue;
                            [myFirstFunctionInputFieldsLeft, myFirstFunctionInputBuilder] = [
                                myFirstFunctionInputNewFieldsLeft - 1,
                                myFirstFunctionInputBuilder
                            ];
                            break;
                        default:
                            throw new typelayerCommonDynamic.DynamicGetException("Unknown Field", { path: dynamicObject.decoderPath() });
                    }
                }
                else {
                    [myFirstFunctionInputFieldsLeft, myFirstFunctionInputBuilder] = [
                        -1,
                        myFirstFunctionInputBuilder
                    ];
                }
            }
            return myFirstFunctionInputBuilder;
            break;
        default:
            throw new typelayerCommonDynamic.DynamicGetException("Unknown Constructor", { path: dynamicObject.decoderPath() });
    }
}
exports.dynamicGetMyFirstFunctionInput = dynamicGetMyFirstFunctionInput;
function dynamicGetObjMyFirstFunctionInput(dynamicObject) {
    let retObj;
    retObj = dynamicGetMyFirstFunctionInput(dynamicObject);
    return retObj;
}
exports.dynamicGetObjMyFirstFunctionInput = dynamicGetObjMyFirstFunctionInput;
function dynamicPutMyFirstFunctionInput(dynamicObject, inputObject) {
    let myFirstFunctionInputFieldMyFirstFunctionInput;
    if (inputObject instanceof
        myFirstPackageServicesMyFirstServiceStructures.MyFirstFunctionInput) {
        let myFirstFunctionInputFieldMyFirstFunctionInput = inputObject;
        dynamicObject.encoderPutStruct("my_first_function_input", 1);
        dynamicObject.encoderPutField("my_first_argument");
        dynamicObject.encoderPutString(myFirstFunctionInputFieldMyFirstFunctionInput.myFirstArgument);
        dynamicObject.encoderPutFieldEnd(true);
        return;
    }
    else {
        throw new Error(`Invalid type ${inputObject}`);
    }
}
exports.dynamicPutMyFirstFunctionInput = dynamicPutMyFirstFunctionInput;
function dynamicGetMyFirstFunctionOutput(dynamicObject) {
    let constructor;
    let myFirstFunctionOutputBuilder;
    let myFirstFunctionOutputFieldName;
    let myFirstFunctionOutputFieldsLeft;
    let myFirstFunctionOutputInit;
    let myFirstFunctionOutputMyFirstReturnValue;
    let myFirstFunctionOutputNewFieldsLeft;
    let numFields;
    [constructor, numFields] = dynamicObject.decoderPopStruct("my_first_function_output");
    switch (constructor) {
        case "my_first_function_output":
            myFirstFunctionOutputInit = myFirstPackageServicesMyFirstServiceStructures.defaultMyFirstFunctionOutput();
            [myFirstFunctionOutputFieldsLeft, myFirstFunctionOutputBuilder] = [
                numFields,
                myFirstFunctionOutputInit
            ];
            while (-1 < myFirstFunctionOutputFieldsLeft) {
                if (0 < myFirstFunctionOutputFieldsLeft) {
                    myFirstFunctionOutputNewFieldsLeft = myFirstFunctionOutputFieldsLeft;
                }
                else {
                    myFirstFunctionOutputNewFieldsLeft = dynamicObject.decoderPopHasMoreStruct();
                }
                if (0 < myFirstFunctionOutputNewFieldsLeft) {
                    myFirstFunctionOutputFieldName = dynamicObject.decoderPopField();
                    switch (myFirstFunctionOutputFieldName) {
                        case "*":
                            throw new typelayerCommonDynamic.DynamicGetException("Constructor must be first field", { path: dynamicObject.decoderPath() });
                            break;
                        case "my_first_return":
                            myFirstFunctionOutputMyFirstReturnValue = myFirstPackageSchemasMyFirstModuleDynamic.dynamicGetMyFirstStructure(dynamicObject);
                            myFirstFunctionOutputBuilder.myFirstReturn = myFirstFunctionOutputMyFirstReturnValue;
                            [
                                myFirstFunctionOutputFieldsLeft,
                                myFirstFunctionOutputBuilder
                            ] = [
                                myFirstFunctionOutputNewFieldsLeft - 1,
                                myFirstFunctionOutputBuilder
                            ];
                            break;
                        default:
                            throw new typelayerCommonDynamic.DynamicGetException("Unknown Field", { path: dynamicObject.decoderPath() });
                    }
                }
                else {
                    [myFirstFunctionOutputFieldsLeft, myFirstFunctionOutputBuilder] = [
                        -1,
                        myFirstFunctionOutputBuilder
                    ];
                }
            }
            return myFirstFunctionOutputBuilder;
            break;
        default:
            throw new typelayerCommonDynamic.DynamicGetException("Unknown Constructor", { path: dynamicObject.decoderPath() });
    }
}
exports.dynamicGetMyFirstFunctionOutput = dynamicGetMyFirstFunctionOutput;
function dynamicGetObjMyFirstFunctionOutput(dynamicObject) {
    let retObj;
    retObj = dynamicGetMyFirstFunctionOutput(dynamicObject);
    return retObj;
}
exports.dynamicGetObjMyFirstFunctionOutput = dynamicGetObjMyFirstFunctionOutput;
function dynamicPutMyFirstFunctionOutput(dynamicObject, inputObject) {
    let myFirstFunctionOutputFieldMyFirstFunctionOutput;
    if (inputObject instanceof
        myFirstPackageServicesMyFirstServiceStructures.MyFirstFunctionOutput) {
        let myFirstFunctionOutputFieldMyFirstFunctionOutput = inputObject;
        dynamicObject.encoderPutStruct("my_first_function_output", 1);
        dynamicObject.encoderPutField("my_first_return");
        myFirstPackageSchemasMyFirstModuleDynamic.dynamicPutMyFirstStructure(dynamicObject, myFirstFunctionOutputFieldMyFirstFunctionOutput.myFirstReturn);
        dynamicObject.encoderPutFieldEnd(true);
        return;
    }
    else {
        throw new Error(`Invalid type ${inputObject}`);
    }
}
exports.dynamicPutMyFirstFunctionOutput = dynamicPutMyFirstFunctionOutput;
function dynamicGetMySecondFunctionInput(dynamicObject) {
    let constructor;
    let mySecondFunctionInputArgOneValue;
    let mySecondFunctionInputArgTwoValue;
    let mySecondFunctionInputBuilder;
    let mySecondFunctionInputFieldName;
    let mySecondFunctionInputFieldsLeft;
    let mySecondFunctionInputInit;
    let mySecondFunctionInputNewFieldsLeft;
    let numFields;
    [constructor, numFields] = dynamicObject.decoderPopStruct("my_second_function_input");
    switch (constructor) {
        case "my_second_function_input":
            mySecondFunctionInputInit = myFirstPackageServicesMyFirstServiceStructures.defaultMySecondFunctionInput();
            [mySecondFunctionInputFieldsLeft, mySecondFunctionInputBuilder] = [
                numFields,
                mySecondFunctionInputInit
            ];
            while (-1 < mySecondFunctionInputFieldsLeft) {
                if (0 < mySecondFunctionInputFieldsLeft) {
                    mySecondFunctionInputNewFieldsLeft = mySecondFunctionInputFieldsLeft;
                }
                else {
                    mySecondFunctionInputNewFieldsLeft = dynamicObject.decoderPopHasMoreStruct();
                }
                if (0 < mySecondFunctionInputNewFieldsLeft) {
                    mySecondFunctionInputFieldName = dynamicObject.decoderPopField();
                    switch (mySecondFunctionInputFieldName) {
                        case "*":
                            throw new typelayerCommonDynamic.DynamicGetException("Constructor must be first field", { path: dynamicObject.decoderPath() });
                            break;
                        case "arg_one":
                            mySecondFunctionInputArgOneValue = myFirstPackageSchemasMyFirstModuleDynamic.dynamicGetMyEnumStructure(dynamicObject);
                            mySecondFunctionInputBuilder.argOne = mySecondFunctionInputArgOneValue;
                            [
                                mySecondFunctionInputFieldsLeft,
                                mySecondFunctionInputBuilder
                            ] = [
                                mySecondFunctionInputNewFieldsLeft - 1,
                                mySecondFunctionInputBuilder
                            ];
                            break;
                        case "arg_two":
                            mySecondFunctionInputArgTwoValue = dynamicObject.decoderPopString();
                            mySecondFunctionInputBuilder.argTwo = mySecondFunctionInputArgTwoValue;
                            [
                                mySecondFunctionInputFieldsLeft,
                                mySecondFunctionInputBuilder
                            ] = [
                                mySecondFunctionInputNewFieldsLeft - 1,
                                mySecondFunctionInputBuilder
                            ];
                            break;
                        default:
                            throw new typelayerCommonDynamic.DynamicGetException("Unknown Field", { path: dynamicObject.decoderPath() });
                    }
                }
                else {
                    [mySecondFunctionInputFieldsLeft, mySecondFunctionInputBuilder] = [
                        -1,
                        mySecondFunctionInputBuilder
                    ];
                }
            }
            return mySecondFunctionInputBuilder;
            break;
        default:
            throw new typelayerCommonDynamic.DynamicGetException("Unknown Constructor", { path: dynamicObject.decoderPath() });
    }
}
exports.dynamicGetMySecondFunctionInput = dynamicGetMySecondFunctionInput;
function dynamicGetObjMySecondFunctionInput(dynamicObject) {
    let retObj;
    retObj = dynamicGetMySecondFunctionInput(dynamicObject);
    return retObj;
}
exports.dynamicGetObjMySecondFunctionInput = dynamicGetObjMySecondFunctionInput;
function dynamicPutMySecondFunctionInput(dynamicObject, inputObject) {
    let mySecondFunctionInputFieldMySecondFunctionInput;
    if (inputObject instanceof
        myFirstPackageServicesMyFirstServiceStructures.MySecondFunctionInput) {
        let mySecondFunctionInputFieldMySecondFunctionInput = inputObject;
        dynamicObject.encoderPutStruct("my_second_function_input", 2);
        dynamicObject.encoderPutField("arg_one");
        myFirstPackageSchemasMyFirstModuleDynamic.dynamicPutMyEnumStructure(dynamicObject, mySecondFunctionInputFieldMySecondFunctionInput.argOne);
        dynamicObject.encoderPutFieldEnd(false);
        dynamicObject.encoderPutField("arg_two");
        dynamicObject.encoderPutString(mySecondFunctionInputFieldMySecondFunctionInput.argTwo);
        dynamicObject.encoderPutFieldEnd(true);
        return;
    }
    else {
        throw new Error(`Invalid type ${inputObject}`);
    }
}
exports.dynamicPutMySecondFunctionInput = dynamicPutMySecondFunctionInput;
function dynamicGetMySecondFunctionOutput(dynamicObject) {
    let constructor;
    let mySecondFunctionOutputBuilder;
    let mySecondFunctionOutputFieldName;
    let mySecondFunctionOutputFieldsLeft;
    let mySecondFunctionOutputInit;
    let mySecondFunctionOutputNewFieldsLeft;
    let mySecondFunctionOutputRetOneElementsLeft;
    let mySecondFunctionOutputRetOneInit;
    let mySecondFunctionOutputRetOneListAcc;
    let mySecondFunctionOutputRetOneListValue;
    let mySecondFunctionOutputRetOneNewFieldsLeft;
    let mySecondFunctionOutputRetOneValue;
    let numFields;
    [constructor, numFields] = dynamicObject.decoderPopStruct("my_second_function_output");
    switch (constructor) {
        case "my_second_function_output":
            mySecondFunctionOutputInit = myFirstPackageServicesMyFirstServiceStructures.defaultMySecondFunctionOutput();
            [mySecondFunctionOutputFieldsLeft, mySecondFunctionOutputBuilder] = [
                numFields,
                mySecondFunctionOutputInit
            ];
            while (-1 < mySecondFunctionOutputFieldsLeft) {
                if (0 < mySecondFunctionOutputFieldsLeft) {
                    mySecondFunctionOutputNewFieldsLeft = mySecondFunctionOutputFieldsLeft;
                }
                else {
                    mySecondFunctionOutputNewFieldsLeft = dynamicObject.decoderPopHasMoreStruct();
                }
                if (0 < mySecondFunctionOutputNewFieldsLeft) {
                    mySecondFunctionOutputFieldName = dynamicObject.decoderPopField();
                    switch (mySecondFunctionOutputFieldName) {
                        case "*":
                            throw new typelayerCommonDynamic.DynamicGetException("Constructor must be first field", { path: dynamicObject.decoderPath() });
                            break;
                        case "ret_one":
                            mySecondFunctionOutputRetOneInit = dynamicObject.decoderPopList();
                            [
                                mySecondFunctionOutputRetOneElementsLeft,
                                mySecondFunctionOutputRetOneListAcc
                            ] = [mySecondFunctionOutputRetOneInit, []];
                            while (-1 < mySecondFunctionOutputRetOneElementsLeft) {
                                if (0 < mySecondFunctionOutputRetOneElementsLeft) {
                                    mySecondFunctionOutputRetOneNewFieldsLeft = mySecondFunctionOutputRetOneElementsLeft;
                                }
                                else {
                                    mySecondFunctionOutputRetOneNewFieldsLeft = dynamicObject.decoderPopHasMoreList();
                                }
                                if (0 < mySecondFunctionOutputRetOneNewFieldsLeft) {
                                    mySecondFunctionOutputRetOneListValue = dynamicObject.decoderPopString();
                                    mySecondFunctionOutputRetOneListAcc.push(mySecondFunctionOutputRetOneListValue);
                                    [
                                        mySecondFunctionOutputRetOneElementsLeft,
                                        mySecondFunctionOutputRetOneListAcc
                                    ] = [
                                        mySecondFunctionOutputRetOneNewFieldsLeft - 1,
                                        mySecondFunctionOutputRetOneListAcc
                                    ];
                                }
                                else {
                                    [
                                        mySecondFunctionOutputRetOneElementsLeft,
                                        mySecondFunctionOutputRetOneListAcc
                                    ] = [-1, mySecondFunctionOutputRetOneListAcc];
                                }
                            }
                            mySecondFunctionOutputRetOneValue = mySecondFunctionOutputRetOneListAcc;
                            mySecondFunctionOutputBuilder.retOne = mySecondFunctionOutputRetOneValue;
                            [
                                mySecondFunctionOutputFieldsLeft,
                                mySecondFunctionOutputBuilder
                            ] = [
                                mySecondFunctionOutputNewFieldsLeft - 1,
                                mySecondFunctionOutputBuilder
                            ];
                            break;
                        default:
                            throw new typelayerCommonDynamic.DynamicGetException("Unknown Field", { path: dynamicObject.decoderPath() });
                    }
                }
                else {
                    [mySecondFunctionOutputFieldsLeft, mySecondFunctionOutputBuilder] = [
                        -1,
                        mySecondFunctionOutputBuilder
                    ];
                }
            }
            return mySecondFunctionOutputBuilder;
            break;
        default:
            throw new typelayerCommonDynamic.DynamicGetException("Unknown Constructor", { path: dynamicObject.decoderPath() });
    }
}
exports.dynamicGetMySecondFunctionOutput = dynamicGetMySecondFunctionOutput;
function dynamicGetObjMySecondFunctionOutput(dynamicObject) {
    let retObj;
    retObj = dynamicGetMySecondFunctionOutput(dynamicObject);
    return retObj;
}
exports.dynamicGetObjMySecondFunctionOutput = dynamicGetObjMySecondFunctionOutput;
function dynamicPutMySecondFunctionOutput(dynamicObject, inputObject) {
    let mySecondFunctionOutputFieldMySecondFunctionOutput;
    if (inputObject instanceof
        myFirstPackageServicesMyFirstServiceStructures.MySecondFunctionOutput) {
        let mySecondFunctionOutputFieldMySecondFunctionOutput = inputObject;
        dynamicObject.encoderPutStruct("my_second_function_output", 1);
        dynamicObject.encoderPutField("ret_one");
        dynamicObject.encoderPutList(mySecondFunctionOutputFieldMySecondFunctionOutput.retOne.length);
        if (0 === mySecondFunctionOutputFieldMySecondFunctionOutput.retOne.length) {
            dynamicObject.encoderPutElementEnd(true);
        }
        else {
            let retOneIndex = 0;
            for (let retOneElem of mySecondFunctionOutputFieldMySecondFunctionOutput.retOne) {
                dynamicObject.encoderPutString(retOneElem);
                dynamicObject.encoderPutElementEnd(retOneIndex ===
                    mySecondFunctionOutputFieldMySecondFunctionOutput.retOne.length - 1);
                retOneIndex++;
            }
        }
        dynamicObject.encoderPutFieldEnd(true);
        return;
    }
    else {
        throw new Error(`Invalid type ${inputObject}`);
    }
}
exports.dynamicPutMySecondFunctionOutput = dynamicPutMySecondFunctionOutput;

File: python/my_first_package/my_first_package/services/my_first_service/dynamic.py
from ...schemas.my_first_module import dynamic as my_first_package_schemas_my_first_module_dynamic
from ...schemas.my_first_module import structures as my_first_package_schemas_my_first_module_structures
from . import structures as my_first_package_services_my_first_service_structures
from typelayer_common import dynamic as typelayer_common_dynamic
from dataclasses import dataclass
from typing import List, Dict, Optional


def dynamic_get_my_first_function_input(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MyFirstFunctionInput:
    
    
    
    
    
    
    
    
    (constructor, num_fields) = dynamic_object.decoder_pop_struct("my_first_function_input")
    if (constructor == "my_first_function_input"):
        my_first_function_input_init = my_first_package_services_my_first_service_structures.default_my_first_function_input()
        (my_first_function_input_fields_left, my_first_function_input_builder) = (num_fields, my_first_function_input_init)
        while ((-1 < my_first_function_input_fields_left)):
            if ((0 < my_first_function_input_fields_left)):
                my_first_function_input_new_fields_left = my_first_function_input_fields_left
            else:
                my_first_function_input_new_fields_left = dynamic_object.decoder_pop_has_more_struct()
            if ((0 < my_first_function_input_new_fields_left)):
                my_first_function_input_field_name = dynamic_object.decoder_pop_field()
                if (my_first_function_input_field_name == "*"):
                    raise typelayer_common_dynamic.DynamicGetException("Constructor must be first field", {"path": dynamic_object.decoder_path()});
                elif (my_first_function_input_field_name == "my_first_argument"):
                    my_first_function_input_my_first_argument_value = dynamic_object.decoder_pop_string()
                    my_first_function_input_builder.my_first_argument = my_first_function_input_my_first_argument_value;
                    
                    (my_first_function_input_fields_left, my_first_function_input_builder) = ((my_first_function_input_new_fields_left - 1), my_first_function_input_builder)
                else:
                        raise typelayer_common_dynamic.DynamicGetException("Unknown Field", {"path": dynamic_object.decoder_path()});
            else:
                (my_first_function_input_fields_left, my_first_function_input_builder) = (-1, my_first_function_input_builder)
        return my_first_function_input_builder
    else:
            raise typelayer_common_dynamic.DynamicGetException("Unknown Constructor", {"path": dynamic_object.decoder_path()});

def dynamic_get_obj_my_first_function_input(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MyFirstFunctionInput:

    ret_obj = dynamic_get_my_first_function_input(dynamic_object)
    return ret_obj

def dynamic_put_my_first_function_input(dynamic_object: typelayer_common_dynamic.Encoder, input_object: my_first_package_services_my_first_service_structures.MyFirstFunctionInput) -> None:

    if isinstance(input_object, my_first_package_services_my_first_service_structures.MyFirstFunctionInput):
        my_first_function_input_field_my_first_function_input = input_object
        dynamic_object.encoder_put_struct("my_first_function_input", 1)
        dynamic_object.encoder_put_field("my_first_argument")
        dynamic_object.encoder_put_string(my_first_function_input_field_my_first_function_input.my_first_argument)
        dynamic_object.encoder_put_field_end(True)
        return;
    else:
        raise Exception(f"Invalid type {input_object}")

def dynamic_get_my_first_function_output(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MyFirstFunctionOutput:
    
    
    
    
    
    
    
    
    (constructor, num_fields) = dynamic_object.decoder_pop_struct("my_first_function_output")
    if (constructor == "my_first_function_output"):
        my_first_function_output_init = my_first_package_services_my_first_service_structures.default_my_first_function_output()
        (my_first_function_output_fields_left, my_first_function_output_builder) = (num_fields, my_first_function_output_init)
        while ((-1 < my_first_function_output_fields_left)):
            if ((0 < my_first_function_output_fields_left)):
                my_first_function_output_new_fields_left = my_first_function_output_fields_left
            else:
                my_first_function_output_new_fields_left = dynamic_object.decoder_pop_has_more_struct()
            if ((0 < my_first_function_output_new_fields_left)):
                my_first_function_output_field_name = dynamic_object.decoder_pop_field()
                if (my_first_function_output_field_name == "*"):
                    raise typelayer_common_dynamic.DynamicGetException("Constructor must be first field", {"path": dynamic_object.decoder_path()});
                elif (my_first_function_output_field_name == "my_first_return"):
                    my_first_function_output_my_first_return_value = my_first_package_schemas_my_first_module_dynamic.dynamic_get_my_first_structure(dynamic_object)
                    my_first_function_output_builder.my_first_return = my_first_function_output_my_first_return_value;
                    
                    (my_first_function_output_fields_left, my_first_function_output_builder) = ((my_first_function_output_new_fields_left - 1), my_first_function_output_builder)
                else:
                        raise typelayer_common_dynamic.DynamicGetException("Unknown Field", {"path": dynamic_object.decoder_path()});
            else:
                (my_first_function_output_fields_left, my_first_function_output_builder) = (-1, my_first_function_output_builder)
        return my_first_function_output_builder
    else:
            raise typelayer_common_dynamic.DynamicGetException("Unknown Constructor", {"path": dynamic_object.decoder_path()});

def dynamic_get_obj_my_first_function_output(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MyFirstFunctionOutput:

    ret_obj = dynamic_get_my_first_function_output(dynamic_object)
    return ret_obj

def dynamic_put_my_first_function_output(dynamic_object: typelayer_common_dynamic.Encoder, input_object: my_first_package_services_my_first_service_structures.MyFirstFunctionOutput) -> None:

    if isinstance(input_object, my_first_package_services_my_first_service_structures.MyFirstFunctionOutput):
        my_first_function_output_field_my_first_function_output = input_object
        dynamic_object.encoder_put_struct("my_first_function_output", 1)
        dynamic_object.encoder_put_field("my_first_return")
        my_first_package_schemas_my_first_module_dynamic.dynamic_put_my_first_structure(dynamic_object, my_first_function_output_field_my_first_function_output.my_first_return)
        dynamic_object.encoder_put_field_end(True)
        return;
    else:
        raise Exception(f"Invalid type {input_object}")

def dynamic_get_my_second_function_input(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MySecondFunctionInput:
    
    
    
    
    
    
    
    
    
    (constructor, num_fields) = dynamic_object.decoder_pop_struct("my_second_function_input")
    if (constructor == "my_second_function_input"):
        my_second_function_input_init = my_first_package_services_my_first_service_structures.default_my_second_function_input()
        (my_second_function_input_fields_left, my_second_function_input_builder) = (num_fields, my_second_function_input_init)
        while ((-1 < my_second_function_input_fields_left)):
            if ((0 < my_second_function_input_fields_left)):
                my_second_function_input_new_fields_left = my_second_function_input_fields_left
            else:
                my_second_function_input_new_fields_left = dynamic_object.decoder_pop_has_more_struct()
            if ((0 < my_second_function_input_new_fields_left)):
                my_second_function_input_field_name = dynamic_object.decoder_pop_field()
                if (my_second_function_input_field_name == "*"):
                    raise typelayer_common_dynamic.DynamicGetException("Constructor must be first field", {"path": dynamic_object.decoder_path()});
                elif (my_second_function_input_field_name == "arg_one"):
                    my_second_function_input_arg_one_value = my_first_package_schemas_my_first_module_dynamic.dynamic_get_my_enum_structure(dynamic_object)
                    my_second_function_input_builder.arg_one = my_second_function_input_arg_one_value;
                    
                    (my_second_function_input_fields_left, my_second_function_input_builder) = ((my_second_function_input_new_fields_left - 1), my_second_function_input_builder)
                elif (my_second_function_input_field_name == "arg_two"):
                    my_second_function_input_arg_two_value = dynamic_object.decoder_pop_string()
                    my_second_function_input_builder.arg_two = my_second_function_input_arg_two_value;
                    
                    (my_second_function_input_fields_left, my_second_function_input_builder) = ((my_second_function_input_new_fields_left - 1), my_second_function_input_builder)
                else:
                        raise typelayer_common_dynamic.DynamicGetException("Unknown Field", {"path": dynamic_object.decoder_path()});
            else:
                (my_second_function_input_fields_left, my_second_function_input_builder) = (-1, my_second_function_input_builder)
        return my_second_function_input_builder
    else:
            raise typelayer_common_dynamic.DynamicGetException("Unknown Constructor", {"path": dynamic_object.decoder_path()});

def dynamic_get_obj_my_second_function_input(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MySecondFunctionInput:

    ret_obj = dynamic_get_my_second_function_input(dynamic_object)
    return ret_obj

def dynamic_put_my_second_function_input(dynamic_object: typelayer_common_dynamic.Encoder, input_object: my_first_package_services_my_first_service_structures.MySecondFunctionInput) -> None:

    if isinstance(input_object, my_first_package_services_my_first_service_structures.MySecondFunctionInput):
        my_second_function_input_field_my_second_function_input = input_object
        dynamic_object.encoder_put_struct("my_second_function_input", 2)
        dynamic_object.encoder_put_field("arg_one")
        my_first_package_schemas_my_first_module_dynamic.dynamic_put_my_enum_structure(dynamic_object, my_second_function_input_field_my_second_function_input.arg_one)
        dynamic_object.encoder_put_field_end(False)
        dynamic_object.encoder_put_field("arg_two")
        dynamic_object.encoder_put_string(my_second_function_input_field_my_second_function_input.arg_two)
        dynamic_object.encoder_put_field_end(True)
        return;
    else:
        raise Exception(f"Invalid type {input_object}")

def dynamic_get_my_second_function_output(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MySecondFunctionOutput:
    
    
    
    
    
    
    
    
    
    
    
    
    
    (constructor, num_fields) = dynamic_object.decoder_pop_struct("my_second_function_output")
    if (constructor == "my_second_function_output"):
        my_second_function_output_init = my_first_package_services_my_first_service_structures.default_my_second_function_output()
        (my_second_function_output_fields_left, my_second_function_output_builder) = (num_fields, my_second_function_output_init)
        while ((-1 < my_second_function_output_fields_left)):
            if ((0 < my_second_function_output_fields_left)):
                my_second_function_output_new_fields_left = my_second_function_output_fields_left
            else:
                my_second_function_output_new_fields_left = dynamic_object.decoder_pop_has_more_struct()
            if ((0 < my_second_function_output_new_fields_left)):
                my_second_function_output_field_name = dynamic_object.decoder_pop_field()
                if (my_second_function_output_field_name == "*"):
                    raise typelayer_common_dynamic.DynamicGetException("Constructor must be first field", {"path": dynamic_object.decoder_path()});
                elif (my_second_function_output_field_name == "ret_one"):
                    my_second_function_output_ret_one_init = dynamic_object.decoder_pop_list()
                    (my_second_function_output_ret_one_elements_left, my_second_function_output_ret_one_list_acc) = (my_second_function_output_ret_one_init, [])
                    while ((-1 < my_second_function_output_ret_one_elements_left)):
                        if ((0 < my_second_function_output_ret_one_elements_left)):
                            my_second_function_output_ret_one_new_fields_left = my_second_function_output_ret_one_elements_left
                        else:
                            my_second_function_output_ret_one_new_fields_left = dynamic_object.decoder_pop_has_more_list()
                        if ((0 < my_second_function_output_ret_one_new_fields_left)):
                            my_second_function_output_ret_one_list_value = dynamic_object.decoder_pop_string()
                            my_second_function_output_ret_one_list_acc.append(my_second_function_output_ret_one_list_value);
                            
                            (my_second_function_output_ret_one_elements_left, my_second_function_output_ret_one_list_acc) = ((my_second_function_output_ret_one_new_fields_left - 1), my_second_function_output_ret_one_list_acc)
                        else:
                            (my_second_function_output_ret_one_elements_left, my_second_function_output_ret_one_list_acc) = (-1, my_second_function_output_ret_one_list_acc)
                    my_second_function_output_ret_one_value = my_second_function_output_ret_one_list_acc
                    my_second_function_output_builder.ret_one = my_second_function_output_ret_one_value;
                    
                    (my_second_function_output_fields_left, my_second_function_output_builder) = ((my_second_function_output_new_fields_left - 1), my_second_function_output_builder)
                else:
                        raise typelayer_common_dynamic.DynamicGetException("Unknown Field", {"path": dynamic_object.decoder_path()});
            else:
                (my_second_function_output_fields_left, my_second_function_output_builder) = (-1, my_second_function_output_builder)
        return my_second_function_output_builder
    else:
            raise typelayer_common_dynamic.DynamicGetException("Unknown Constructor", {"path": dynamic_object.decoder_path()});

def dynamic_get_obj_my_second_function_output(dynamic_object: typelayer_common_dynamic.Decoder) -> my_first_package_services_my_first_service_structures.MySecondFunctionOutput:

    ret_obj = dynamic_get_my_second_function_output(dynamic_object)
    return ret_obj

def dynamic_put_my_second_function_output(dynamic_object: typelayer_common_dynamic.Encoder, input_object: my_first_package_services_my_first_service_structures.MySecondFunctionOutput) -> None:

    if isinstance(input_object, my_first_package_services_my_first_service_structures.MySecondFunctionOutput):
        my_second_function_output_field_my_second_function_output = input_object
        dynamic_object.encoder_put_struct("my_second_function_output", 1)
        dynamic_object.encoder_put_field("ret_one")
        dynamic_object.encoder_put_list(len(my_second_function_output_field_my_second_function_output.ret_one))
        if ((0 == len(my_second_function_output_field_my_second_function_output.ret_one))):
            dynamic_object.encoder_put_element_end(True)
        else:
            ret_one_index = 0;
            for ret_one_elem in my_second_function_output_field_my_second_function_output.ret_one:
                dynamic_object.encoder_put_string(ret_one_elem)
                dynamic_object.encoder_put_element_end((ret_one_index == (len(my_second_function_output_field_my_second_function_output.ret_one) - 1)))
                
                ret_one_index += 1
            
        dynamic_object.encoder_put_field_end(True)
        return;
    else:
        raise Exception(f"Invalid type {input_object}")